File: datasets/rs-keep-code_davinci_001_temp_0.2/HumanEval_158_find_max.rs
/// Write a function that accepts a list of strings.
/// The list contains different words. Return the word with maximum number
/// of unique characters. If multiple strings have maximum number of unique
/// characters, return the one which comes first in lexicographical order.
/// find_max(["name", "of", "string"]) == "string"
/// find_max(["name", "enam", "game"]) == "enam"
/// find_max(["aaaaaaa", "bb" ,"cc"]) == ""aaaaaaa"
fn find_max(words: Vec<String>) -> String {
    let mut max_len = 0;
    let mut max_word = String::new();
    for word in words {
        let mut chars: HashSet<char> = HashSet::new();
        for c in word.chars() {
            chars.insert(c);
        }
        if chars.len() > max_len {
            max_len = chars.len();
            max_word = word;
        }
    }
    max_word

}

fn main() {
    let candidate = find_max;
    assert_eq!(candidate(vec![String::from("name"), String::from("of"), String::from("string")]), String::from("string"));
    assert_eq!(candidate(vec![String::from("name"), String::from("enam"), String::from("game")]), String::from("enam"));
    assert_eq!(candidate(vec![String::from("aaaaaaa"), String::from("bb"), String::from("cc")]), String::from("aaaaaaa"));
    assert_eq!(candidate(vec![String::from("abc"), String::from("cba")]), String::from("abc"));
    assert_eq!(candidate(vec![String::from("play"), String::from("this"), String::from("game"), String::from("of"), String::from("footbott")]), String::from("footbott"));
    assert_eq!(candidate(vec![String::from("we"), String::from("are"), String::from("gonna"), String::from("rock")]), String::from("gonna"));
    assert_eq!(candidate(vec![String::from("we"), String::from("are"), String::from("a"), String::from("mad"), String::from("nation")]), String::from("nation"));
    assert_eq!(candidate(vec![String::from("this"), String::from("is"), String::from("a"), String::from("prrk")]), String::from("this"));
    assert_eq!(candidate(vec![String::from("b")]), String::from("b"));
    assert_eq!(candidate(vec![String::from("play"), String::from("play"), String::from("play")]), String::from("play"));
}

File: datasets/R-keep-code_davinci_001_temp_0.2/HumanEval_158_find_max.R
# Write a function that accepts a list of strings.
# The list contains different words. Return the word with maximum number
# of unique characters. If multiple strings have maximum number of unique
# characters, return the one which comes first in lexicographical order.
# find_max(["name", "of", "string"]) == "string"
# find_max(["name", "enam", "game"]) == "enam"
# find_max(["aaaaaaa", "bb" ,"cc"]) == ""aaaaaaa"
find_max <- function(words) {
    max_len <- 0
    max_word <- ""
    for (word in words) {
        if (length(unique(word)) > max_len) {
            max_len <- length(unique(word))
            max_word <- word
        }
    }
    max_word
}
find_max(c("name", "of", "string"))
find_max(c("name", "enam", "game"))
find_max(c("aaaaaaa", "bb" ,"cc"))


test_humaneval <- function() {
candidate <- find_max
    if(!identical(candidate(list('name', 'of', 'string')), 'string')){quit('no', 1)}
    if(!identical(candidate(list('name', 'enam', 'game')), 'enam')){quit('no', 1)}
    if(!identical(candidate(list('aaaaaaa', 'bb', 'cc')), 'aaaaaaa')){quit('no', 1)}
    if(!identical(candidate(list('abc', 'cba')), 'abc')){quit('no', 1)}
    if(!identical(candidate(list('play', 'this', 'game', 'of', 'footbott')), 'footbott')){quit('no', 1)}
    if(!identical(candidate(list('we', 'are', 'gonna', 'rock')), 'gonna')){quit('no', 1)}
    if(!identical(candidate(list('we', 'are', 'a', 'mad', 'nation')), 'nation')){quit('no', 1)}
    if(!identical(candidate(list('this', 'is', 'a', 'prrk')), 'this')){quit('no', 1)}
    if(!identical(candidate(list('b')), 'b')){quit('no', 1)}
    if(!identical(candidate(list('play', 'play', 'play')), 'play')){quit('no', 1)}
}
test_humaneval()